tap_typeform/client.py

```diff
@@ -190,3 +107,60 @@ def raise_for_error(response):
 
         except (ValueError, TypeError):
             raise TypeformError(error) from None
+
+class Client(object):
+    BASE_URL = 'https://api.typeform.com'
+
+    def __init__(self, config):
+        self.token = 'Bearer ' + config.get('token')
+        self.metric = config.get('metric')
+        self.session = requests.Session()
+        self.page_size = MAX_RESPONSES_PAGE_SIZE
+        self.form_page_size = FORMS_PAGE_SIZE
+        self.get_page_size(config)
+
+        # Set and pass request timeout to config param `request_timeout` value.
+        config_request_timeout = config.get('request_timeout')
+        if config_request_timeout and float(config_request_timeout):
+            self.request_timeout = float(config_request_timeout)
+        else:
+            self.request_timeout = REQUEST_TIMEOUT # If value is 0,"0","" or not passed then it set default to 300 seconds.
+
+    def get_page_size(self, config):
+        """
+        This function will get page size from config,
+        and will return the default value if invalid page size is given.
+        """
+        page_size = config.get('page_size')
+        try:
+            if page_size is None:
+                pass
+            elif int(float(page_size)) > 0:
+                self.page_size = int(float(page_size))
+                self.form_page_size = min(self.form_page_size, int(float(page_size)))
+            else:
+                raise Exception
```

Comment:
Is it possible to add a message in Exception?

Reply:
Here we have just raised th exception, the message is added in the catch block

Reply:
in this scenario what user need to interpret about the error?

Reply:
It will raise an error and catch it and there it will throw the error with a proper invalid page size error message.

Reply:
This should still be rewritten because Exceptions are not control flow mechanisms. There are a few reasons for this, but most salient for me is long-term maintenance and readable code. Because an exception isn't bound by the written control flow, and instead relies on runtime state to determine where it goes, future refactoring in code where exceptions are used as control flow can cause things to get out of hand and behave in unexpected ways.

It may look fine in certain cases, but the potential negative impact of keeping this as a valid pattern outweighs the effort required to just use `if` statements and scope the outer `try/catch` just to where it is needed (the `int(float(page_size))` call).

Reply:
Thank you for the detailed explanation and I definitely agree with your point. Updated the code to use `if-else` instead of `try/catch` and used the try-catch only for typecasting conversions.